Mathlib/FieldTheory/IsAlgClosed/Basic.lean

```diff
@@ -208,6 +208,17 @@ theorem IntermediateField.eq_bot_of_isAlgClosed_of_isAlgebraic {k K : Type*} [Fi
   obtain ⟨y, hy⟩ := IsAlgClosed.algebraMap_surjective_of_isIntegral (k := k) (⟨x, hx⟩ : L)
   exact ⟨y, congr_arg (algebraMap L K) hy⟩
 
+lemma Polynomial.isCoprime_iff_aeval_ne_zero_of_isAlgClosed (K : Type v) [Field K] [IsAlgClosed K]
```

Comment:
Can you generalize this without assuming `K` to be algebraically closed but taking `a` in an algebraic closure? (We can also keep this version of course)

Reply:
You mean something like this?
```
lemma Polynomial.isCoprime_iff_aeval_ne_zero_of_isAlgClosure (K : Type v) [Field K] [Algebra k K] [IsAlgClosure k K]
    (p q : k[X]) : IsCoprime p q ↔ ∀ a : K, aeval a p ≠ 0 ∨ aeval a q ≠ 0 :=
  @Polynomial.isCoprime_iff_aeval_ne_zero_of_isAlgClosed _ _ _ _ (IsAlgClosure.alg_closed k) _ _ _
```
That seems to be a specialisation rather than a generalisation?

Unless you mean the fixed `AlgebraicClosure`?

Reply:
How are you avoiding that coprimality is preserved by field extensions?

Reply:
I'm not quite sure what you mean. We also have `Polynomial.isCoprime_iff_aeval_ne_zero` which is the version that only talks about field extensions, but I don't know how to make the proof work for algebraic closures.

Reply:
I am sorry, you're lemma already says what I wanted.